src/components/Wizard/Patterns/StatefulWizardPattern.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import polyfill from 'react-lifecycles-compat';
+
+import WizardPattern from './WizardPattern';
+import { wizardStepShape } from './WizardPatternConstants';
+import { noop, propOrState, bindMethods, excludeKeys } from '../../../index';
+
+/**
+ * StatefulWizardPattern - the Stateful Wizard Pattern component.
+ */
+class StatefulWizardPattern extends React.Component {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return {
+      activeStepIndex: propOrState(nextProps, prevState, 'activeStepIndex')
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { activeStepIndex: 0 };
+    bindMethods(this, ['onStepChanged']);
```

Comment:
Optional since this is used everywhere else.  onStepChange can be written like
```javascript
onStepChanged = (newStepIndex) =>
```
Writing it like this will "autobind" the method to the class instance.